numpy/core/tests/test_umath.py

```diff
@@ -249,6 +249,29 @@ def test_division_int(self):
         assert_equal(x // 100, [0, 0, 0, 1, -1, -1, -1, -1, -2])
         assert_equal(x % 100, [5, 10, 90, 0, 95, 90, 10, 0, 80])
 
+    @pytest.mark.parametrize("input_dtype",
+            [np.int8, np.int16, np.int32, np.int64])
+    def test_division_int_boundary(self, input_dtype):
+        class ListWithDiv(list):
+            def __floordiv__(self, divisor):
+                return [i//divisor for i in self]
+
+        iinfo = np.iinfo(input_dtype)
+
+        # Create array with min, 25th percentile, 0, 75th percentile, max
+        arr = ListWithDiv([iinfo.min, iinfo.min//2, 0, iinfo.max//2, iinfo.max])
+        dividends = [iinfo.min, iinfo.min//2, iinfo.max//2, iinfo.max]
+        a = np.array(arr, dtype = input_dtype)
+
+        for dividend in dividends:
+            div_a = a // dividend
```

Comment:
I would like a test for `b = a.copy(); b //= dividend` just be pedantic about also testing the case where the operation is done in-place.